content/client-server-api/modules/end_to_end_encryption.md

```diff
@@ -1587,7 +1598,11 @@ that they can decrypt future messages encrypted using this session. An
 `m.room_key` events sent by other devices in order to decrypt their
 messages.
 
-When a client is updating a Megolm session (room key) in its store, the client MUST ensure:
+When a client receives a Megolm session, the client MUST ensure that the
+session was received via a channel that ensures authenticity of the messages.
```

Comment:
Do we need to complicate here, rather than just demanding it's via Olm (for now)? My concern is that we don't explain what kind of authenticity properties we are demanding of the channel here.

Reply:
In theory, a client could use some other type of channel that's custom to the client.  Then again, this PR already says that the `m.room_key` event "is encrypted as an `m.room.encrypted` event using Olm", so I guess I'll just change it to say that it must be sent over Olm.